src/dcam.camera.c

```diff
@@ -257,12 +258,15 @@ set_output_trigger__handler(HDCAM h,
                          ? DCAMPROP_OUTPUTTRIGGER_POLARITY__NEGATIVE
                          : DCAMPROP_OUTPUTTRIGGER_POLARITY__POSITIVE;
         used_lines[line] = 1;
-        array_prop_write(i32, h, DCAM_IDPROP_OUTPUTTRIGGER_KIND, line, kind);
+        CHECK(
+          array_prop_write(i32, h, DCAM_IDPROP_OUTPUTTRIGGER_KIND, line, kind));
 
-        array_prop_write(
-          i32, h, DCAM_IDPROP_OUTPUTTRIGGER_SOURCE, line, source);
+        if (kind == DCAMPROP_OUTPUTTRIGGER_KIND__PROGRAMABLE) {
```

Comment:
Consider moving this to `set_output_triggering`? Or the body inside the condition. We discussed this live and agreed that it's fine as is because that would need some code repetition and this is not critical.